Formula/u/uhdm.rb

```diff
@@ -0,0 +1,74 @@
+class Uhdm < Formula
+  include Language::Python::Virtualenv
+
+  desc "Universal Hardware Data Model, modeling of the SystemVerilog Object Model"
+  homepage "https://github.com/chipsalliance/UHDM"
+  url "https://github.com/chipsalliance/UHDM/archive/refs/tags/v1.74.tar.gz"
+  sha256 "92cefa641610457772c91d45864e3f4d65bfa0bf300886dc3067a561358a8eed"
+  license "Apache-2.0"
+  head "https://github.com/chipsalliance/UHDM.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "python@3.11" => :build
+  depends_on "six" => :build
+  depends_on "pkg-config" => :test
+  depends_on "capnp"
+
+  resource "orderedmultidict" do
+    url "https://files.pythonhosted.org/packages/53/4e/3823a27d764bb8388711f4cb6f24e58453e92d6928f4163fdb01e3a3789f/orderedmultidict-1.0.1.tar.gz"
+    sha256 "04070bbb5e87291cc9bfa51df413677faf2141c73c61d2a5f7b26bea3cd882ad"
+  end
+
+  def install
+    venv = virtualenv_create(buildpath/"venv", "python3.11")
+    resources.each do |r|
+      venv.pip_install r
+    end
+    # Build shared library
+    system "cmake", "-S", ".", "-B", "build_shared",
+      "-DBUILD_SHARED_LIBS=ON",
```

Comment:
Is this definitely needed? Seems to duplicate `build_shared`

Reply:
this is absolutely needed, this is the CMake flag to build shared libraries as opposed to static libraries.